### DISCUSSION TITLE
Unable to connect openid authentication

### DISCUSSION THREAD

**Discussion by 321nikhi** [keycloak/keycloak#21148] [Q&A] [answered]
I am trying to connect with Google authentication & openid connect. 
Getting error when keycloak trying to authenticate with identity provider. 
giving error " Unexpected error when authenticating with identity provider".
my kuberenetes cluster don't have internet acccess.
![keycloak error](https://github.com/keycloak/keycloak/assets/110227103/ccc775ba-c5bc-40d5-a07a-617c337b59b2)
![ssl-handshake](https://github.com/keycloak/keycloak/assets/110227103/e037f86b-bf28-4d2d-a30a-e312c7cc88f3)

Your suggestions would be great help.

**Comment by sschu** [comment, marked as answer]
Keycloak does need Internet access to retrieve a token from Google. Thats how OIDC works.

**Reply by 321nikhi** [reply]
Hi sschu 
thankyou for your responce,
we are publically accessing keycloak through application gateway. first authentication call to Identity provider through domain name is succesfull. we are getting responce from identity to keycloak it redirects to the endpoint which we are provided to identity provider. facing isue with call back, Is there ant way to make that call private or keycloak should not callback.

**Reply by 321nikhi** [reply]
when internate was provided to the kuberenetes cluster it was working fine. it is not working when we removed cluster internate access.

**Reply by sschu** [reply]
As I said, Internet access is necessary.
